exercises/bowling/example.go

```diff
@@ -0,0 +1,126 @@
+// Package bowling implements scoring for the game of bowling.
+package bowling
+
+import "errors"
+
+const testVersion = 1
+
+var (
+	ErrNegativeRollIsInvalid        = errors.New("Negative roll is invalid")
+	ErrPinCountExceedsPinsOnTheLane = errors.New("Pin count exceeds pins on the lane")
+	ErrPrematureScore               = errors.New("Score cannot be taken until the end of the game")
+	ErrCannotRollAfterGameOver      = errors.New("Cannot roll after game is over")
+)
+
+// Game records the data to track a game's progress.
+type Game struct {
+	rolls       [21]int // storage for the rolls (10th frame gets up to 3)
+	nRolls      int     // counts the rolls accumulated.
+	nFrames     int     // counts completed frames, up to 10.
+	rFrameStart int     // tracks the starting roll of each frame.
+}
+
+// NewGame returns a fresh zero-valued game struct.
+func NewGame() *Game {
+	return &Game{}
+}
+
+// Roll records one roll for a bowling frame with 'pins' knocked down.
+// An error is possible depending on pin value and previous rolls.
+func (g *Game) Roll(pins int) error {
+	// Validate pin count on roll.
+	if pins > 10 {
```

Comment:
Ah, I had overlooked that because bowling has the "fixed" constants.  Good point about 10 frames vs. 10 pins.  Will fix it up.